src/nanodbc.cpp

```diff
@@ -881,80 +881,42 @@ class connection::connection_impl
 
     void* native_env_handle() const { return env_; }
 
-    string_type dbms_name() const
+    string_type get_string_info(short info_type) const
     {
-        NANODBC_SQLCHAR name[255] = {0};
+        NANODBC_SQLCHAR value[255] = {0};
         SQLSMALLINT length(0);
         RETCODE rc;
         NANODBC_CALL_RC(
             NANODBC_FUNC(SQLGetInfo),
             rc,
             conn_,
-            SQL_DBMS_NAME,
-            name,
-            sizeof(name) / sizeof(NANODBC_SQLCHAR),
+            info_type,
+            value,
+            sizeof(value) / sizeof(NANODBC_SQLCHAR),
             &length);
         if (!success(rc))
             NANODBC_THROW_DATABASE_ERROR(conn_, SQL_HANDLE_DBC);
-        return string_type(&name[0], &name[strarrlen(name)]);
+        return string_type(&value[0], &value[strarrlen(value)]);
```

Comment:
No it is only intended to return null-terminated character strings, we could presumably have corresponding functions for the other types (or at least a `get_integer_info()`) that would handle all of the integer types.